Code/Tools/FBuild/FBuildCore/Helpers/VSProjectGenerator.cpp

```diff
@@ -86,6 +88,48 @@ void VSProjectGenerator::AddFiles( const Array< AString > & files )
     guid.Format( "{%08x-6c94-4f93-bc2a-7f5284b7d434}", CRC32::Calc( projectNameNormalized ) );
 }
 
+// IsFilenameAHeaderFile
+//------------------------------------------------------------------------------
+static inline const bool IsFilenameAHeaderFile( const AString & fileName )
+{
+    // TODO: Add support for other header types?
+    return ( fileName.EndsWith( 'h' ) || fileName.EndsWith( ".hpp" ) || fileName.EndsWith( ".hxx" ) || fileName.EndsWith( ".inl" ) );
+}
+
+static inline const bool ShouldWeSkipThisPlatform( const AString & platformIncludePattern, const AString & platform )
+{
+    // If we have a .PlatformIncludePattern filter defined, and this platform doesn't match it, return true [So we can skip emitting this platform to the .vcxproj]
+    const char *thisPlatformIncludePattern = platformIncludePattern.Get();
+    ASSERT( (thisPlatformIncludePattern != nullptr) && (platformIncludePattern != "") );
+    return ( (!platformIncludePattern.IsEmpty()) && (!platform.Find( thisPlatformIncludePattern )) );
+}
+
+static inline const char* GetPossiblyAliasedPlatformStringInUppercase( const AString & targetArchitectureAlias, const AString & platform )
+{
+    // Handle Platform Aliases - these define new platforms and alias existing platforms to them
+    if( !targetArchitectureAlias.IsEmpty() )
+    {
+        const char *thisPlatformAlias = targetArchitectureAlias.Get();
+        ASSERT( (thisPlatformAlias != nullptr) && (targetArchitectureAlias != "") );
+        AStackString<>							thisPlatformAliasString( thisPlatformAlias );
+                                                thisPlatformAliasString.ToUpper();
+        const char *thisPlatformAliasUpper =	thisPlatformAliasString.Get();
+        return thisPlatformAliasUpper;
+    }
+    return platform.Get();
+}
+
+static inline const char* GetPossiblyAliasedPlatformString( const AString & targetArchitectureAlias, const AString & platform )
+{
+    // Handle Platform Aliases - these define new platforms and alias existing platforms to them
+    if( !targetArchitectureAlias.IsEmpty() )
+    {
+        ASSERT( (targetArchitectureAlias.Get() != nullptr) && (targetArchitectureAlias != "") );
```

Comment:
This assert can never fire - AString never returns nullptr, and the IsEmpty() checkes means it can never be == ""

Reply:
Thanks - I'll remove all the `== ""` and `nullptr` asserts - they were there because I was writing code quickly and wanted to know if my assumptions were correct or not about how your string classes worked, without taking the time to read all of the code. I think I planned to come back and review it later and it seems I never did. Sorry :(